### ISSUE TITLE
[New build - DEFAULT] 21w50a: Space Police Patrols

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#4373]
Space Police patrol ships have been added! They will enforce the Safe Zone around stations.
There are two small patrol ships for now but more will be added soon - thanks to the builder Popka Akula and 5URG3!
Some quality of life improvements to the blueprint loading experience have also been added.

New Features and Improvements:
 - Space Police: Space Police ships are now patrolling around the stations
 - Space Police: If a ship breaks the safe zone rules Space Police patrol ships will warn then strike back
 - Space Police: Station Safe zone HUD element will reflect the player current hostile status
 - NPC: Various improvements have been made to the NPCs hostile behavior
 - Game Performance: Garbage Collection greatly reduced in idle and during tutorial phase: it means that small hiccups have been reduced
 - Blueprint Load: filter added
 - Blueprint Load: Can now filter on Workshop or Local blueprints
 - Blueprint Load: A new Blueprint Preview side window has been added: it shows a larger picture and some info on the blueprint 
 (you will need to re-export your blueprint to see those info)
 - Physics engine: depenetration logic was updated. This fixes the twitchy handheld drop
 - Physics engine: smarter collision sfx
 - World Generation: Less stations are being generated on startup, leading to faster load time
 - Tutorial: various improvements
 
 Community Suggestion:
 #4337 Blueprint information preview

Hotfixes:
#4346 Issue with handheld with no primary handle
#4348 Handheld: issue when there is only a single handheld at Secondary
#4352 Spawned Blueprint can collide with large ship
#4353 Spawned Pirate Blueprint not aggressive
Various handheld issues
Various Hull Point and repair issues

**Comment by tsunamayo** [comment]
Build is up!

**Comment by cloudrambler** [comment]
I just loaded a previous world into 21w50a and it did not progress past the artwork page.

[Player.log](https://github.com/tsunamayo/Starship-EVO/files/7738672/Player.log)

**Comment by tsunamayo** [comment]
Okay could anyone else report if they have issues or not?

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo The preview window of the currently active grid is broken

![image](https://user-images.githubusercontent.com/63756057/146641967-6375783d-1527-4049-9fd7-6bd964e6af8c.png)

The Safe Zone status UI indicator is bugged because I warped to another station and the police did not attack me there:

![image](https://user-images.githubusercontent.com/63756057/146642292-0650070b-4bbd-494d-ab6e-d704492b6bfa.png)

5URGE's police ship has its logic panel exposed by default 😳 

![image](https://user-images.githubusercontent.com/63756057/146641425-a95d05b4-79ef-4a20-8c46-306610e0cdb1.png)

I think it's a good progression to introduce stronger waves of bigger and bigger police hunter ships that will chase you across the star system because right now, if you destroy both interceptors, you are free to wreak havoc upon the station they were defending.

And I see that NPCs can finally fire the mouse-operated weapons! Great success!

But the best update IMO was the blueprint filter. I was waiting so long for this and it's such a great feature! Thank you!

**Comment by Kaiser-Indrasil** [comment]
And I've noticed the patrol interceptors doing loops:

![WEEOO](https://user-images.githubusercontent.com/63756057/146643401-c2ad3166-7f75-4149-aa26-45bfd27235e9.gif)

**Comment by Uncle-Ulty** [comment]
welp, the police aren't reacting at all... I destroyed one of the patrol ships, and nothing happens

![image](https://user-images.githubusercontent.com/73253472/146650926-7a5b5b67-3cec-4154-9a41-55d13480b3af.png)

was it supposed to as like GTA police? You break the rules and they go after you?

**Comment by Kaiser-Indrasil** [comment]
@Uncle-Ulty No, for now only the 2 interceptors assigned to the station attack you, but only if you are around the station. They don't pursue you across the system nor do other police ships from other stations engage you, though I wish the game sent some autogenerated pursuit ships after you, much like in GTA, with more and bigger ships sent your way the more aggressive you are.

**Comment by tsunamayo** [comment]
Yes I will send bigger ship after you kill the first two. There will be a bounty system thereafter - you will have to pay to clean your record, or be hunted everywhere you go. I am just not to that part yet!
@Kaiser-Indrasil What did you do? Did you unspawn your ship by any chance?
Hunted status is per station for now.
I will also make sure that if you attack a police ship the other one will look at you!

Thanks

**Comment by Kaiser-Indrasil** [comment]
No, i didn't. I just attacked the two ships, flew in the same ship to the other station, and the status was the same

**Comment by Leidz** [comment]
I noticed this on the prefab tab, the picture are gone since the update

![gone](https://user-images.githubusercontent.com/47060013/146668962-6aafaebc-bcdb-4b14-bc2f-edbe5ba325f6.PNG)

**Comment by tsunamayo** [comment]
I am doing a roll back of that update as some players are reporting an issue on startup.

**Comment by Uncle-Ulty** [comment]
Hi @tsunamayo.

about the Police ships, I believe you added the ones with fewer bricks, right?

If bricks and CE aren't a big problem (which I doubt), I'd like invite you to check my recently updated starwatch ship.

https://steamcommunity.com/sharedfiles/filedetails/?id=2688595081
![image](https://user-images.githubusercontent.com/73253472/147132337-b223e0b3-54c8-4275-8ab9-a878949990e7.png)
